### ISSUE TITLE
Suggestion: Prefix pugixml specific CMake options to avoid name collision with other libraries

### ISSUE THREAD

**Issue by bjornblissing** [zeux/pugixml#374]
When including pugixml as a submodule to our own project the CMake options and cached variables may result in name collisions.

https://github.com/zeux/pugixml/blob/c4bae8d8185858435bdf22cd98ba7e58e67d9041/CMakeLists.txt#L18-L20

https://github.com/zeux/pugixml/blob/c4bae8d8185858435bdf22cd98ba7e58e67d9041/CMakeLists.txt#L26-L30

My suggestion is to rename the following options and cached variables:
```
BUILD_SHARED_AND_STATIC_LIBS 
BUILD_DEFINES
STATIC_CRT
```

to
```
PUGIXML_BUILD_SHARED_AND_STATIC_LIBS 
PUGIXML_BUILD_DEFINES
PUGIXML_STATIC_CRT
```

The variable `BUILD_SHARED_LIBS` is a CMake built-in variable and probably should be removed. Alternatively defined as:
```cmake
cmake_dependent_option(PUGIXML_BUILD_SHARED_LIBS 
  "Build shared instead of static library" OFF
  "BUILD_SHARED_LIBS" ON)
```

The options below are not created if pugixml is built from outside the pugixml source directory (i.e. another project), so there is no pressing need to prefix those. 
```
USE_VERSIONED_LIBDIR
USE_POSTFIX
BUILD_TESTS
```

**Comment by hsdk123** [comment]
This would be great

**Comment by zeux** [comment]
Fixed by #399.
